aquasecurity/tracee
# trigger: move event triggering logic out of derive

## Initial Checklist

- [x] I have discussed this change with another maintainer.

## Description (git log)

```
Author: Nadav Strahilevitz <nadav.strahilevitz@aquasec.com>
Date:   Sun Aug 14 12:51:05 2022 +0000

    trigger: move event triggering logic out of derive
    
    This was done in order to:
    1. Give the mechanism separation a starting point.
    2. Clean up the derive package later.
    
    Triggered events now have a context map interface which stores the
    invoking events for later use. This is used down the pipeline
    in processEvents.
```

Fixes: #issue_number

## Type of change


## How Has This Been Tested?

1. Install a rootkit such as (diamorphine)[https://github.com/m0nad/Diamorphine]
2. Start tracee-ebpf with `sudo ./dist/tracee-ebpf -t e=hooked_syscalls,hooked_seq_ops`
3. Load diamorphine
4. Check the events are outputted the same as before
5. Unload diamorphine

Unit tests were added under:
`pkg/events/trigger/context_test.go`

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.


## Git Log Checklist:

My commits logs have:
